File: Assignment5/Library.java
import edu.princeton.cs.algs4.In;
public class Library
{
  public static void main(String[] args)
  { 
    In in = new In(args[0]);
    int takeOut = in.readInt();
    int books = in.readInt();
    int day = in.readInt();
    int count = 1;
    while(books >= 0)
    { 
      if (day == 6 || day == 7)
        books -= count;
      else
        books = books + takeOut - count;      
      count++;
      if (day == 7) 
        day = 1;
      else
        day++;
    }
    int res = 0;
    if (count != 1)
      res = count - 2;
    System.out.print(res);
  }
}

File: Assignment5/Library/Library/Library.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main(int argc, char* argv[])
{
	if (argc < 2)
	{
		cout << "no input data\n";
		return 0;
	}

	ifstream in(argv[1]);
	if (!in)
	{
		cout << "Cannot open file.\n";
		return 0;
	}

	int takeOut;
	int books;
	int day;
	in >> takeOut;
	in >> books;
	in >> day;
	int count = 1;
	while (books >= 0)
	{
		if (day == 6 || day == 7)
			books -= count;
		else
			books = books + takeOut - count;
		count++;
		day = (day + 1) % 7 + 1;
	}
	in.close();
	int res = 0;
	if (count != 1)
		res = count - 2;
	cout << res;
	char ch;
	cin >> ch;
	return 1;
}